src/wui/load_or_save_game.h

```diff
@@ -77,29 +79,43 @@ class LoadOrSaveGame {
 	/// Show confirmation window and delete the selected file(s)
 	void clicked_delete();
 
+	void change_directory_to(std::string& directory);
+
 private:
 	/// Returns the filename for the table entry at 'index'
-	const std::string get_filename(int index) const;
-
+	const std::string get_filename(uint32_t index) const;
+	/// Returns the savegame for the table entry at 'index'
+	const SavegameData& get_savegame(uint32_t index) const;
 	/// Formats the current table selection as a list of filenames with savedate information.
 	const std::string filename_list_string() const;
 	/// Formats a given table selection as a list of filenames with savedate information.
 	const std::string filename_list_string(const std::set<uint32_t>& selections) const;
 
 	/// Reverse default sort order for save date column
-	bool compare_date_descending(uint32_t, uint32_t) const;
+	bool compare_save_time(uint32_t, uint32_t) const;
+	bool compare_map_name(uint32_t, uint32_t) const;
 
 	UI::Panel* parent_;
 	UI::Box* table_box_;
-	UI::Table<uintptr_t const> table_;
 	FileType filetype_;
-	bool show_filenames_;
-	bool localize_autosave_;
+
+	std::unique_ptr<SavegameTable> table_;
```

Comment:
The UI::Panel class does all the memory management for our UI classes, including tables.

So, although we want to move to unique_ptr everywhere else, you could simply use `SavegameTable table_`; here and your code should still be safe.

Reply:
Since SavegameTable is an abstract type, I need to use a pointer, right? Do you want me to use a raw pointer SavegameTable* table_?

Reply:
Yes, this would be sufficient, because memory management for our UI toolkit is done by `UI::Panel`. Generally, we want to use `unique_ptr`, but the UI classes are a exception.

BTW when constructing the table, you can use a switch statement, that will be more efficient than the if/else.